src/cargo/ops/cargo_test.rs

```diff
@@ -158,6 +164,19 @@ fn run_doc_tests(
             .arg("--crate-name")
             .arg(&target.crate_name());
 
+        if doctest_xcompile {
+            p.arg("-Zunstable-options");
+            p.arg("--enable-per-target-ignores");
+        }
+
+        p.arg("--target").arg(&compilation.target);
```

Comment:
To be conservative, can this continue to be in the `doctest_xcomple` block above?